gtsam/discrete/DiscreteFactorGraph.cpp

```diff
@@ -217,12 +214,21 @@ namespace gtsam {
   std::pair<DiscreteConditional::shared_ptr, DiscreteFactor::shared_ptr>  //
   EliminateDiscrete(const DiscreteFactorGraph& factors,
                     const Ordering& frontalKeys) {
-    DiscreteFactor::shared_ptr product = factors.scaledProduct();
+    gttic(product);
+    DiscreteFactor::shared_ptr product = factors.product();
+    gttoc(product);
 
     // sum out frontals, this is the factor on the separator
     gttic(sum);
     DiscreteFactor::shared_ptr sum = product->sum(frontalKeys);
     gttoc(sum);
+    
+    // Normalize/scale to prevent underflow.
+    gttic(scale);
+    DiscreteFactor::shared_ptr denominator = sum->max(sum->size());
+    product = product->operator/(denominator);
+    sum = sum->operator/(denominator);
```

Comment:
I don't get it. Why does this work?
We went from doing `product/max(product)` to `product/max(sum)`, and in both cases `sum->scale()`. The denominators will cancel later, but  haven't we now changed what was computed?